systems-management/nisysmgmt.yml

```diff
@@ -0,0 +1,2092 @@
+          type: string
+          x-example: test.ping
+        - in: query
+          name: systemId
+          description: A string defining the system ID.
+          type: string
+          x-example: VMware_Virtual_Platform--SN-VMware-56_4d_30_26_4f_c9_5c_05-b7_68_d1_cd_99_6e_22_16--MAC-02-01-02-03-04-08
+      responses:
+        200:
+          description: OK.
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Job'
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/systems:
+    get:
+      operationId: GetSystems
+      x-ni-privilege: Read
+      x-ni-operation: getSystems
+      tags: [systems]
+      summary: Get systems information
+      description: Get the systems information cached on the server.
+      parameters:
+        - in: query
+          name: id
+          description: A string defining the system ID.
+          type: string
+          x-example: VMware_Virtual_Platform--SN-VMware-56_4d_30_26_4f_c9_5c_05-b7_68_d1_cd_99_6e_22_16--MAC-02-01-02-03-04-08
+      responses:
+        200:
+          description: OK.
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/ManagedSystem'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+        default:
+          $ref: '#/responses/Error'
+  /v1/register-systems:
+    post:
+      operationId: RegisterSystems
+      x-ni-privilege: Write
+      x-ni-operation: manageSystems
+      tags: [systems]
+      summary: Register systems
+      description: Make a request to register systems. This is an asynchronous request. A successful status code informs that the request has been acknowledged and not that all the registration procedures have completed successfully.
+      parameters:
+        - in: body
+          name: RegisterRequest
+          description: Registration information.
+          required: true
+          schema:
+            $ref: '#/definitions/RegisterSystemsRequest'
+      responses:
+        200:
+          description: OK.
+          schema:
+            $ref: '#/definitions/RegisterSystemsResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+        400:
+          $ref: '#/responses/BadRequest'
+        default:
+          $ref: '#/responses/Error'
+  /v1/remove-systems:
+    post:
+      operationId: UnregisterSystems
+      x-ni-privilege: Write
+      x-ni-operation: manageSystems
+      tags: [systems]
+      summary: Unregister Systems
+      description: Request to unregister systems.
+      parameters:
+        - in: body
+          name: UnregisterSystemsRequest
+          description: Request data to unregister systems.
+          required: true
+          schema:
+            $ref: '#/definitions/UnregisterSystemsRequest'
+      responses:
+        200:
```

Comment:
Seems this could also return partial success, if only a subset of systems were unregistered.